### PR TITLE
Add listAuctionHistory rpc

### PR DESCRIPTION
**Pull request by jingyi2811**
#### What kind of PR is this?:

/kind feature

#### What this PR does / why we need it:
Add `listauctionhistory` rpc

#### Which issue(s) does this PR fixes?:

Fixes part of #488

### CODE REVIEW

**Review by canonbrother — DISMISSED** [review]

File: packages/jellyfish-api-core/__tests__/category/loan/listAuctionHistory.test.ts [hunk 1]
```diff
@@ -0,0 +1,592 @@
...
+        const data3 = data.find(d => d.vaultId === vaultId3)
+        expect(data3).toBeDefined()
+
+        const data4 = data.find(d => d.vaultId === vaultId4)
+        expect(data4).toBeDefined()
+      }
+    })
+  })
+
+  describe('listAuctionHistory with pagination', () => { // For pagination, always filter by 'all'.
```
Comment (surangap) [inline packages/jellyfish-api-core/__tests__/category/loan/listAuctionHistory.test.ts:484]: `For pagination, always filter by 'all'.` that is actually not true. you can paginate the results which are filtered by `mine ` or `owner`. and it's real world use case that we filter by `mine ` or `owner` and get the results paginated.
Can add a test for that too.
note that `owner` is filtration and all fields in `ListAuctionHistoryPagination` are for pagination.
Here also database index has `height` as first. So it's crucial that we pass the intended `height` that the pagination should start.

from cpp side
```
inline void SerializationOp(Stream& s, Operation ser_action) {

        if (ser_action.ForRead()) {
            READWRITE(WrapBigEndian(blockHeight));
            blockHeight = ~blockHeight;
        } else {
            uint32_t blockHeight_ = ~blockHeight;
            READWRITE(WrapBigEndian(blockHeight_));
        }

        READWRITE(owner);
        READWRITE(vaultId);
        READWRITE(index);
    }
```

Comment (jingyi2811) [reply packages/jellyfish-api-core/__tests__/category/loan/listAuctionHistory.test.ts]: Added the test units for 
- 'mine' and address.
- height + vaultId
- height + index

Comment (surangap) [reply packages/jellyfish-api-core/__tests__/category/loan/listAuctionHistory.test.ts]: Added some minor refactor to pagination tests.

**Review by surangap — APPROVED** [review]

**Review by canonbrother — APPROVED** [review]

**Review by fuxingloh — APPROVED** [review]

File: packages/jellyfish-api-core/src/category/loan.ts [hunk 2]
```diff
@@ -396,6 +396,24 @@ export class Loan {
...
+   *
+   * @param {string} [owner] address or reserved word : mine / all (Default to mine)
+   * @param {ListAuctionHistoryPagination} pagination
+   * @param {number} [pagination.maxBlockHeight] Maximum block height
+   * @param {string} [pagination.vaultId] Vault Id
+   * @param {number} [pagination.index] Auction index
+   * @param {number} [pagination.limit = 100]
+   * @return {Promise<ListAuctionHistoryDetail>}
+   */
+  async listAuctionHistory (owner: string = 'mine', pagination?: ListAuctionHistoryPagination): Promise<ListAuctionHistoryDetail[]> {
```
Comment (ivan-zynesis) [inline packages/jellyfish-api-core/src/category/loan.ts:411]: hmm, I know the vault id here is solely sorting index. But feel like list auction history as per vault is more useful for UI like `GET /auctions/:vaultId`

**Pull request merged**